src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -0,0 +1,334 @@
+        String predecessorName = dependentClass.getSimpleName();
+        boolean isAllowed = userReadFilterNames.contains(predecessorName)
+                || (toolDefaultReadFilters.get(predecessorName) != null);
+        if (isAllowed) {
+            // keep track of the ones we allow so we can validate later that they
+            // weren't subsequently disabled
+            requiredPredecessors.add(predecessorName);
+        }
+        return isAllowed;
+    }
+
+    /**
+     * Pass back the list of ReadFilter instances that were actually seen on the
+     * command line in the same order they were specified. This list does not
+     * include the tool defaults.
+     */
+    @Override
+    public List<ReadFilter> getAllInstances() {
+        // Add the instances in the order they were specified on the command line
+        // (use the order of userReadFilterNames list).
+        //
+        // NOTE: it's possible for the userReadFilterNames list to contain one or more
+        // names for which there are no corresponding instances in the readFilters list.
+        // This happens when the user specifies a filter name on the command line that's
+        // already included in the toolDefault list, since in that case the descriptor
+        // uses the tool-supplied instance and doesn't add a separate one to the
+        // readFilters list, but the name from the command line still appears in
+        // userReadFilterNames. In that case, we don't include the tool's instance in the
+        // list returned by this method since it will be merged in later by the merge method.
+        final ArrayList<ReadFilter> filters = new ArrayList<>(userReadFilterNames.size());
+        userReadFilterNames.forEach(s -> {
+            ReadFilter rf = readFilters.get(s);
+            if (rf != null) {
+                filters.add(rf);
+            }
+        });
+        return filters;
+    }
+
+    // Return the allowable values for readFilterNames/disableReadFilter
+    @Override
+    public Set<String> getAllowedValuesForDescriptorArgument(final String longArgName) {
+        if (longArgName.equals(StandardArgumentDefinitions.READ_FILTER_LONG_NAME) ||
+                longArgName.equals(disableReadFilterArgName)) {
+            return readFilters.keySet();
+        }
+        throw new IllegalArgumentException("Allowed values request for unrecognized string argument: " + longArgName);
+    }
+
+    /**
+     * Validate the list of arguments and reduce the list of read filters to those
+     * actually seen on the command line. This is called by the command line parser
+     * after all arguments have been parsed.
+     */
+    @Override
+    public void validateArguments() {
+        // throw if any filter is both enabled *and* disabled by the user
+        final Set<String> enabledAndDisabled = new HashSet<>(userReadFilterNames);
+        enabledAndDisabled.retainAll(disableFilters);
+        enabledAndDisabled.forEach(
```

Comment:
It's a bit strange to have this as a forEach since it can really only happen once.